app/models/compute_resources/foreman/model/libvirt.rb

```diff
@@ -275,7 +275,9 @@ def create_volumes(args)
         (volumes = args[:volumes]).each do |vol|
           vol.name = "#{args[:prefix]}-disk#{volumes.index(vol) + 1}"
           vol.capacity = "#{vol.capacity}G" unless vol.capacity.to_s.end_with?('G')
-          vol.allocation = "#{vol.allocation}G" unless vol.allocation.to_s.end_with?('G')
+          if vol.allocation.match(/^\d+/) && vol.allocation.to_s.end_with?('G')
+            vol.allocation = "#{vol.allocation}G"
```

Comment:
You will append 'G' if the allocation already ends with it, resulting in:

![](https://imgur.com/FOK0BQ6.png)